examples/webserver/threadpool/src/main/java/io/helidon/examples/webserver/threadpool/GreetService.java

```diff
@@ -0,0 +1,212 @@
+ *
+ * Get default greeting message:
+ * curl -X GET http://localhost:8080/greet
+ *
+ * Get greeting message for Joe:
+ * curl -X GET http://localhost:8080/greet/Joe
+ *
+ * Change greeting
+ * curl -X PUT -H "Content-Type: application/json" -d '{"greeting" : "Howdy"}' http://localhost:8080/greet/greeting
+ *
+ * The message is returned as a JSON object
+ */
+
+public class GreetService implements Service {
+
+    /**
+     * The config value for the key {@code greeting}.
+     */
+    private final AtomicReference<String> greeting = new AtomicReference<>();
+
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());
+
+    private static final Logger LOGGER = Logger.getLogger(GreetService.class.getName());
+
+    private static ExecutorService myThreadPool;
+
+    GreetService(Config config) {
+        greeting.set(config.get("app.greeting").asString().orElse("Ciao"));
+
+        // Build a thread pool using the configuration
+        myThreadPool = ThreadPoolSupplier.builder().config(config.get("application-thread-pool")).build().get();
+        if (myThreadPool instanceof ContextAwareExecutorService) {
+            ExecutorService wrapped = ((ContextAwareExecutorService) myThreadPool).unwrap();
+            LOGGER.info("Application thread pool: " + wrapped.toString());
+        }
+    }
+
+    /**
+     * A service registers itself by updating the routing rules.
+     * @param rules the routing rules.
+     */
+    @Override
+    public void update(Routing.Rules rules) {
+        rules
+            .get("/", this::getDefaultMessageHandler)
+            .get("/slowly/{name}", this::getMessageSlowlyHandler)
+            .get("/{name}", this::getMessageHandler)
+            .put("/greeting", this::updateGreetingHandler);
+
+    }
+
+    /**
+     * Return a worldly greeting message.
+     * @param request the server request
+     * @param response the server response
+     */
+    private void getDefaultMessageHandler(ServerRequest request, ServerResponse response) {
+        sendResponse(response, "World");
+    }
+
+    /**
+     * Return a greeting message using the name that was provided.
+     * @param request the server request
+     * @param response the server response
+     */
+    private void getMessageHandler(ServerRequest request, ServerResponse response) {
+        String name = request.path().param("name");
+        sendResponse(response, name);
+    }
+
+    /**
+     * Slowly Return a greeting message using the name that was provided.
+     * @param request the server request
+     * @param response the server response
+     */
+    private void getMessageSlowlyHandler(ServerRequest request, ServerResponse response) {
+
+        String name = request.path().param("name");
+
+        // One way to pass data to new thread is to use Context
+        Contexts.context().ifPresentOrElse(ctx -> ctx.register("NAME_PARAM", name + "_from_context"), () -> {
```

Comment:
You are guaranteed to run in a context here - you can use `request.context()` to obtain an instance of it.

Reply:
Fixed